Add certificate renewal procedures for containerized deployments (#4856)
theforeman/foreman-documentation: guides/common/modules/proc_renewing-self-signed-certificates-on-server.adoc

:_mod-docs-content-type: PROCEDURE

[id="renewing-self-signed-certificates-on-{project-context}-server"]
= Renewing self-signed certificates on {ProjectServer}

[role="_abstract"]
Renew expired self-signed server and client certificates on {ProjectServer}.
You can also set a custom validity period for the renewed certificates.

[NOTE]
====
To renew the certificate authority (CA) certificate, follow a separate process to avoid disrupting connections to existing hosts.
ifndef::containerized[]
For more information, see xref:planning-for-self-signed-ca-certificate-renewal[].
endif::[]
====

.Procedure

> . Renew server and client certificates:

Reviewer: ```suggestion
* Renew server and client certificates:
```

+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy --certificate-renew
----
+
To set a custom validity period, append `--certificate-validity-days _My_Days_` with the validity period in days for the new certificate.

.Verification

> . Verify the new certificate validity period:

Reviewer: IMO on the other PR, you went with an unordered list. Refs line 29 and 37.

:_mod-docs-content-type: PROCEDURE

[id="renewing-self-signed-certificates-on-{project-context}-server"]
= Renewing self-signed certificates on {ProjectServer}

[role="_abstract"]
Renew expired self-signed server and client certificates on {ProjectServer}.
You can also set a custom validity period for the renewed certificates.

[NOTE]
====
To renew the certificate authority (CA) certificate, follow a separate process to avoid disrupting connections to existing hosts.
ifndef::containerized[]
For more information, see xref:planning-for-self-signed-ca-certificate-renewal[].
endif::[]
====

.Procedure
* Renew server and client certificates:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy --certificate-renew
----
+

> To set a custom validity period, append `--certificate-validity-days _My_Days_` with the validity period in days for the new certificate.

Reviewer 2: Don't you think we should also include `--certificate-ca-validity-days` for the CA certificate, as that could be part of the append as well?

Also, one more point: if we are generating the default certificate, should we mention the default expiry duration here, such as:
Defaults: both are 7300 days (approximately 20 years)?

Reviewer 2: We cannot set `--certificate-ca-validity-days` with a new certificate, as it does not update afterward. This is a one-time setting for the CA validity period, and changing it later will not apply in this case.

Reviewer 2: On the above point, I feel we should be clearer about this. Below are my suggestions:

I think we should first mention how it works with the default certificate configuration. By default, it does not require any additional parameters, and the `foremanctl deploy` command sets the default certificate expiry period to 7300 days.

Second, we should explain how to configure custom certificate validity during the initial deployment using --`certificate-ca-validity-days` and `--certificate-validity-days`. This can be done with the following command, which also covers the CA certificate validity:

`foremanctl deploy --certificate-ca-validity-days <DAYS> --certificate-validity-days <DAYS>`

Third, we should describe how `--certificate-renew` works. With this option, only the server and client certificates can be renewed, while the CA certificate cannot be renewed through this command.

I think this would provide a much clearer explanation of certificate validity and renewal behavior.

Reviewer 2: @jafiala 
For `Renewing certificates` section, it looks good. However, are we also planning to cover the remaining parts or covered somewhere, as I mentioned above?

Author: For CA certs, that's what the note above is for:

```
[NOTE]
====
To renew the certificate authority (CA) certificate, follow a separate process to avoid disrupting connections to existing hosts.
ifndef::containerized[]
For more information, see xref:planning-for-self-signed-ca-certificate-renewal[].
endif::[]
====
```

When Administering Foreman is ready for containerized, this will lead to the CA-specific steps.

Will add the defaults, good call.

.Verification
* In your browser, go to your {ProjectServer} login page, for example, `\https://{foreman-example-com}`, and inspect the certificate in the browser.
This is typically displayed as a shield, padlock, or tune icon next to the address bar depending on your browser.

> 

Reviewer: ```suggestion
```

:_mod-docs-content-type: PROCEDURE

[id="renewing-self-signed-certificates-on-{project-context}-server"]
= Renewing self-signed certificates on {ProjectServer}

[role="_abstract"]
Renew expired self-signed server and client certificates on {ProjectServer}.
You can also set a custom validity period for the renewed certificates.

[NOTE]
====
To renew the certificate authority (CA) certificate, follow a separate process to avoid disrupting connections to existing hosts.
ifndef::containerized[]
For more information, see xref:planning-for-self-signed-ca-certificate-renewal[].
endif::[]
====

.Procedure
. Renew server and client certificates:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy --certificate-renew
----
. Optional: Renew certificates and configure a custom validity period:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy --certificate-renew \
--certificate-validity-days _DAYS_
----
+

> Replace _DAYS_ with the number of days the new certificates should remain valid.

Reviewer: Maybe this could be simplified further:

```suggestion
+
If you want to set a custom validity period, append `--certificate-validity-days _My_Days_` with the number of days the new certificates should remain valid.
```

IMO this could replace lines 25-33. If you apply this, line 19 should become an unordered list.

:_mod-docs-content-type: PROCEDURE

[id="renewing-self-signed-certificates-on-{project-context}-server"]
= Renewing self-signed certificates on {ProjectServer}

[role="_abstract"]
Renew expired self-signed server and client certificates on {ProjectServer}.
You can also set a custom validity period for the renewed certificates.

[NOTE]
====
To renew the certificate authority (CA) certificate, follow a separate process to avoid disrupting connections to existing hosts.
ifndef::containerized[]
For more information, see xref:planning-for-self-signed-ca-certificate-renewal[].
endif::[]
====

.Procedure
. Renew server and client certificates:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy --certificate-renew
----
. Optional: Renew certificates and configure a custom validity period:
+
[options="nowrap", subs="+quotes,attributes"]
----

> # {foremanctl} deploy --certificate-renew \

Reviewer: ```suggestion
# {foremanctl} deploy \
```

Looks like a copy and paste error.

Author: Yes, but I'd like to keep --certificate-renew on the same line here to align with the previous step.

Reviewer: non-blocking: Usually, if there are multiple options/arguments, we split them into separate lines.

:_mod-docs-content-type: PROCEDURE

> [id="renewing-self-signed-certificates-on-{project-context}_{context}"]

Reviewer: ```suggestion
[id="renewing-self-signed-certificates-on-{project-context}-server"]
```

= Renewing self-signed certificates on {ProjectServer}

[role="_abstract"]
Renew expired self-signed server and client certificates on {ProjectServer}.

[NOTE]
====
To renew the certificate authority (CA) certificate, follow a separate process to avoid disrupting connections to existing hosts.
For more information, see xref:planning-for-self-signed-ca-certificate-renewal[].
====

.Procedure
. Renew server and client certificates:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy --certificate-renew
----

> 

Reviewer: ```suggestion
```

. Optional: Renew certificates and configure a custom validity period:
+
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy \
--certificate-renew \
--certificate-validity-days _DAYS_
----
+
Replace _DAYS_ with the number of days the new certificates should remain valid.
+
.Example: Renew certificates with 1-year validity
[options="nowrap", subs="+quotes,attributes"]
----
# {foremanctl} deploy \
--certificate-renew \
--certificate-validity-days 365

> ----

Reviewer: I do not think that this adds any value.

Author: ```suggestion
```

.Verification
. Verify the new certificate validity period:
+
[options="nowrap", subs="+quotes,attributes"]
----

> # openssl x509 -in /etc/pki/katello/certs/katello-apache.crt -noout -dates

Reviewer 3: ```suggestion
# echo | openssl s_client -connect 127.0.0.1:443 2>/dev/null | openssl x509 -noout -dates
```

----
+

> The output displays the certificate's `notBefore` and `notAfter` dates, where `notBefore` is the date when the certificate becomes valid and `notAfter` is the expiration date.

Reviewer: same as above.

Author: ```suggestion
The output displays the `notBefore` and `notAfter` dates of the certificate, where `notBefore` is the date when the certificate becomes valid and `notAfter` is the date when it expires.
```

> 

Reviewer: ```suggestion
```

> . In your browser, go to your {ProjectServer} login page, for example, `\https://{foreman-example-com}`. and inspect the certificate in the browser.

Reviewer: ```suggestion
. In your browser, go to your {ProjectServer} login page, for example, `\https://{foreman-example-com}`, and inspect the certificate in the browser.
```

typo
